File: datamodel/src/main/java/datamodel/clinical/TumorStage.java
package datamodel.clinical;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public enum TumorStage {
    I(null),
    II(null),
    IIA(TumorStage.II),
    IIB(TumorStage.II),
    III(null),
    IIIA(TumorStage.III),
    IIIB(TumorStage.III),
    IIIC(TumorStage.III),
    IV(null);

    @Nullable
    private final TumorStage category;

    TumorStage(@Nullable final TumorStage category) {
        this.category = category;
    }

    @Nullable
    public TumorStage category() {
        return category;
    }

    @NotNull
    public String display() {
        return this.toString();
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/datamodel/clinical/TumorStage.kt
package com.hartwig.actin.datamodel.clinical

import com.hartwig.actin.datamodel.Displayable

enum class TumorStage(val category: TumorStage?) : Displayable {
    I(null),
    II(null),
    IIA(II),
    IIB(II),
    III(null),
    IIIA(III),
    IIIB(III),
    IIIC(III),
    IV(null);

    override fun display(): String {
        return this.toString()
    }
}
